ethereum/client/src/lib.rs

```diff
@@ -73,51 +79,133 @@ pub enum TxStatus {
     Failed,
 }
 
-impl Contracts {
+#[derive(Clone)]
+pub struct EthApi {
+    contracts: Contracts<ProviderType, Http<Client>, Ethereum>,
+}
```

Comment:
```suggestion
pub struct EthApi(Contracts<ProviderType, Http<Client>, Ethereum>)
```

Reply:
And implement `Deref` for `EthApi` to not duplicate `Contracts` API

Reply:
Done, but Deref brings out ethereum types those are not really necessary

Reply:
Discussed in DM